src/js/components/CosmosErrorMessage.js

```diff
@@ -41,25 +46,58 @@ class CosmosErrorMessage extends React.Component {
   }
 
   getDetails() {
-    let {data} = this.props.error;
-    if (!data || !data.errors) {
-      // No errors
+    let {error} = this.props;
+    if (!error) {
       return null;
     }
+
+    // Check if this is a plain string
+    if (typeof error === 'string') {
+      return [error];
+
     // Check if we should additionally append
     // the error details as an unordered list
+    } else if (error.data && error.data.errors) {
```

Comment:
Not a fan of `else-if`s, they make it hard to reason about and it is unnecessary in this case, because you return in the block above. Could we do this instead:

```js
  getDetails() {
    let {error} = this.props;
    // No errors
    if (!error || !error.data || !error.data.errors) {
      return null;
    }

    // Check if this is a plain string
    if (typeof error === 'string') {
      return [error];

    }

    // Check for invalid errors format
    if (!Array.isArray(error.data.errors)) {
      return [String(error.data.errors)];
    }

    // Get an array of array of errors for every individual path
    let errorsDetails = error.data.errors.map(function (errorDetail) {

      // Bail early on unexpected error object format
      if (!errorDetail) {
        return [];
      }
      if (typeof errorDetail !== 'object') {
        return [String(errorDetail)];
      }

      // Extract details
      let {path = '/', errors = []} = errorDetail;
      if (!errors || !Array.isArray(errors)) {
        return [];
      }
      return errors.map(function (error) {
        return (ErrorPaths[path] || path)+'.'+error;
      });
    });

    // Flatten elements in array and return
    return errorsDetails.reduce(function (a, b) {
      return a.concat(b);
    });
  }
```